File: Easy/2999_Check_if_Strings_Can_be_Made_Equal_With_Operations_I.java
/*
 * Problem: Check if Strings Can be Made Equal With Operations I
 * Problem ID: 2999
 * Difficulty: Easy
 * Language: Java
 * Runtime: 1 ms
 * Memory: 44.7 MB
 * Synced From: LeetCode
 * Date: 2026-08-12
 */

class Solution {
    public boolean canBeEqual(String s1, String s2) {
        
        boolean evenMatch = (s1.charAt(0) == s2.charAt(0) && s1.charAt(2) == s2.charAt(2)) ||
                            (s1.charAt(0) == s2.charAt(2) && s1.charAt(2) == s2.charAt(0));

        
        boolean oddMatch  = (s1.charAt(1) == s2.charAt(1) && s1.charAt(3) == s2.charAt(3)) ||
                            (s1.charAt(1) == s2.charAt(3) && s1.charAt(3) == s2.charAt(1));

      
        return evenMatch && oddMatch;
    }
}

File: Easy/2999_Check_if_Strings_Can_be_Made_Equal_With_Operations_I.cpp
/*
 * Problem: Check if Strings Can be Made Equal With Operations I
 * Problem ID: 2999
 * Difficulty: Easy
 * Language: C++
 * Runtime: 2 ms
 * Memory: 8.3 MB
 * Synced From: LeetCode
 * Date: 2026-08-16
 */

class Solution {
public:
    bool canBeEqual(string s1, string s2) {
         bool evenMatch = (s1[0] == s2[0] && s1[2] == s2[2]) ||
                            (s1[0] == s2[2] && s1[2] == s2[0]);

        
        bool oddMatch  = (s1[1] == s2[1] && s1[3] == s2[3]) ||
                            (s1[1] == s2[3] && s1[3] == s2[1]);

      
        return evenMatch && oddMatch;
    }
        
    
};